### PR TITLE
Various voicecall manager fixes

### PR DESCRIPTION
**Pull request by MerlijnWajer**
This should fix all known issues with voicecall manager, except for the 'randomly not getting audio' bug -- but I think we have ascertained this problem is likely in the kernel, and not in voicecall-manager, so that will get fixed another way.

### CODE REVIEW

File: src/modules/store-rtcom.c [hunk 1]
```diff
@@ -234,7 +236,7 @@ static GList *get_messages_for_contact(Contact *contact, unsigned int limit)
 		if(!backend)
 			return NULL;
 
-		char *localid = g_strconcat("sphone/", backend->name, "/", "sphone", NULL);
+		char *localid = g_strdup(backend->name);
```
Comment (IMbackK) [inline src/modules/store-rtcom.c:239]: Isent this going to break all existing entries? why this change in the first place? How is "ofono" better than sphone/ofono/sphone here?

Comment (MerlijnWajer) [reply src/modules/store-rtcom.c:239]: Good point. In voicecall manager this will be `ring/tel/account0`. If you want it to be `sphone/ofono/sphone` then I think this would ideally be set in the comm-ofono module.

It has to be `ring/tel/account0` for VCM cellular. We don't want to put "sphone/" in front of that, or after it.

Comment (MerlijnWajer) [reply src/modules/store-rtcom.c:239]: I could create a postinst query to update old entries if you would like that.

Comment (IMbackK) [reply src/modules/store-rtcom.c:239]: the problem here is that the storage backend requiring the comm backend name to be something specific/ special is unacceptable to me

Comment (IMbackK) [reply src/modules/store-rtcom.c:239]: more to the point: the backend->name is really supposed to only be a string for the gui to show and the previous code was on fairly thin ice here anyhow but requiring the backend name to be some very specific sheme just to satisfy one potential storage backend is not reasonable

Comment (MerlijnWajer) [reply src/modules/store-rtcom.c:239]: Okay, can you come up with some solutions that work for you and work for Maemo? We're not going to change all the other applications to log as `sphone/.../sphone`.

Comment (MerlijnWajer) [reply src/modules/store-rtcom.c:239]: The `ring/tel/account0` part seems very reasonable to me, and the `sphone/ofono/sphone` thing was literally made to mimick this, so I don't see why you can't have this be set like this in `comm-ofono`. On top of that, if you want to a more user-friendly string in the UI, then split out the UI string and the database logging string - that only seems reasonable to me.

Comment (MerlijnWajer) [reply src/modules/store-rtcom.c:239]: The `local_uid` and `remote_uid` are defined for rtcom in the sense that they need to be in a specific format, not adhering to it kind of defeats the purpose of the rtcom backend. This is also why you made the ofono backend log as `sphone/ofono/sphone` (connection manager, protocol, account name). I just removed the forced sphone prefix since it doesn't below in the `local_uid` in rtcom if the backend provides a correct TP local uid.

To me the solution seems pretty simple: have a string that represents what is it is called in the UI and a string for logging purposes. That will also make it better for the user so pick between different XMPP, SIP and cellular accounts for making a call.

Comment (IMbackK) [reply src/modules/store-rtcom.c:239]: maybe so, but this patch is for sure NAK, please remove it from this pr, not least because it also completely breaks call and sms history in sphone as both of those expect the central string in // to correspond to the backend name

Comment (IMbackK) [reply src/modules/store-rtcom.c:239]: > so I don't see why you can't have this be set like this in `comm-ofono`

The reason for this is because the // scheme is a rtcom-el specific thing and previously it was set so in store-rtcom which is fine, leaking these storage specific details through sphone into unrelated modules is however not fine, which is what this would be.

Comment (MerlijnWajer) [reply src/modules/store-rtcom.c:239]: >maybe so, but this patch is for sure NAK, please remove it from this pr, not least because it also completely breaks call and sms history in sphone as both of those expect the central string in // to correspond to the backend name

This comes down a poor design decision to always insert "sphone/" before the backend name append "/sphone" after it, I don't see the point of changing the PR because then it won't work on Maemo (which is the primary reason I work on any of this) so I'm not inclined to do that. If you don't want to change the backend name in comm-ofono for to be backwards compatible with your previous design decision here, then maybe I can change the PR to have store-rtcom check if the backend name is "ofono" and then prepend "sphone/" before the backend name and append "/sphone" at the end, and otherwise use the backend name.

Comment (IMbackK) [reply src/modules/store-rtcom.c:239]: You appear to not get the fundamental issues here:

1. this pr is broken because you changed the way sphone stores events but not how it reads them back.
2. forcing the backend name to some specific scheme to appease store-rtcom is unacceptable. It is unacceptable for ofono and its unacceptable that store-rtcom forces the vcms module name to be ring/tel/account0
3. even if we change this to allow vcm to be called "tel" and ofono to be called "ofono" or whatever i am not convinced logging as ring/tel/account0 is a good idea either, because its a lie, we are not ring we are sphone, it should be sphone/vcm/account0 or something like that maybe, if that is a problem because some stuff requires local_uid to be some telepathy specific string, then that confirms my suspicion that rtcom-el is unstable as a generic non-telepathy bound logging framework and it must be changed, maybe by adding an optional telepathy-id field or whatever.
